### ISSUE TITLE
0.11.19 SQLite Db Errors

### ISSUE THREAD

**Issue by Jiriteach** [alexjustesen/speedtest-tracker#701]
Seeing db errors with the latest release. Since I used Watchtower - it pulled down the latest 0.11.19 and now the container wont come back up. Have multiple deployments effected.

Assuming this is related to the SQLite changes? How does one fix this? Was a working deployment - only update was the container.

```
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(272): Illuminate\\Database\\SQLiteConnection->__construct()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(76): Illuminate\\Database\\Connectors\\ConnectionFactory->createConnection()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(50): Illuminate\\Database\\Connectors\\ConnectionFactory->createSingleConnection()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(152): Illuminate\\Database\\Connectors\\ConnectionFactory->make()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(101): Illuminate\\Database\\DatabaseManager->makeConnection()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(210): Illuminate\\Database\\DatabaseManager->connection()
#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(98): Illuminate\\Cache\\CacheManager->createDatabaseDriver()
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(61): Illuminate\\Cache\\CacheManager->resolve()
#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(72): Illuminate\\Cache\\CacheManager->store()
#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheServiceProvider.php(36): Illuminate\\Cache\\CacheManager->driver()
#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(908): Illuminate\\Cache\\CacheServiceProvider->Illuminate\\Cache\\{closure}()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(795): Illuminate\\Container\\Container->build()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(937): Illuminate\\Container\\Container->resolve()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): Illuminate\\Foundation\\Application->resolve()
#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(922): Illuminate\\Container\\Container->make()
#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1454): Illuminate\\Foundation\\Application->make()
#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(236): Illuminate\\Container\\Container->offsetGet()
#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(207): Illuminate\\Support\\Facades\\Facade::resolveFacadeInstance()
#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(347): Illuminate\\Support\\Facades\\Facade::getFacadeRoot()
#23 /var/www/html/app/Providers/RouteServiceProvider.php(29): Illuminate\\Support\\Facades\\Facade::__callStatic()
#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\\Providers\\RouteServiceProvider->boot()
#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#27 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call()
#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1010): Illuminate\\Container\\Container->call()
#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(991): Illuminate\\Foundation\\Application->bootProvider()
#31 [internal function]: Illuminate\\Foundation\\Application->Illuminate\\Foundation\\{closure}()
#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(992): array_walk()
#33 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\\Foundation\\Application->boot()
#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(261): Illuminate\\Foundation\\Bootstrap\\BootProviders->bootstrap()
#35 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(447): Illuminate\\Foundation\\Application->bootstrapWith()
#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(199): Illuminate\\Foundation\\Console\\Kernel->bootstrap()
#37 /var/www/html/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#38 {main}
[previous exception] [object] (Illuminate\\Database\\SQLiteDatabaseDoesNotExistException(code: 0): Database file at path [/var/www/html/database/database.sqlite] does not exist. Ensure this is an absolute path to the database. at /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connectors/SQLiteConnector.php:34)

[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(221): Illuminate\\Database\\Connectors\\SQLiteConnector->connect()
#1 [internal function]: Illuminate\\Database\\Connectors\\ConnectionFactory->Illuminate\\Database\\Connectors\\{closure}()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(1292): call_user_func()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(574): Illuminate\\Database\\Connection->getPdo()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(788): Illuminate\\Database\\Connection->Illuminate\\Database\\{closure}()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(755): Illuminate\\Database\\Connection->runQueryCallback()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(581): Illuminate\\Database\\Connection->run()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Schema/Builder.php(398): Illuminate\\Database\\Connection->statement()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/SQLiteConnection.php(36): Illuminate\\Database\\Schema\\Builder->enableForeignKeyConstraints()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(272): Illuminate\\Database\\SQLiteConnection->__construct()
#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(76): Illuminate\\Database\\Connectors\\ConnectionFactory->createConnection()
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(50): Illuminate\\Database\\Connectors\\ConnectionFactory->createSingleConnection()
#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(152): Illuminate\\Database\\Connectors\\ConnectionFactory->make()
#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(101): Illuminate\\Database\\DatabaseManager->makeConnection()
#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(210): Illuminate\\Database\\DatabaseManager->connection()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(98): Illuminate\\Cache\\CacheManager->createDatabaseDriver()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(61): Illuminate\\Cache\\CacheManager->resolve()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(72): Illuminate\\Cache\\CacheManager->store()
#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Cache/CacheServiceProvider.php(36): Illuminate\\Cache\\CacheManager->driver()
#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(908): Illuminate\\Cache\\CacheServiceProvider->Illuminate\\Cache\\{closure}()
#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(795): Illuminate\\Container\\Container->build()
#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(937): Illuminate\\Container\\Container->resolve()
#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): Illuminate\\Foundation\\Application->resolve()
#23 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(922): Illuminate\\Container\\Container->make()
#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1454): Illuminate\\Foundation\\Application->make()
#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(236): Illuminate\\Container\\Container->offsetGet()
#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(207): Illuminate\\Support\\Facades\\Facade::resolveFacadeInstance()
#27 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(347): Illuminate\\Support\\Facades\\Facade::getFacadeRoot()
#28 /var/www/html/app/Providers/RouteServiceProvider.php(29): Illuminate\\Support\\Facades\\Facade::__callStatic()
#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\\Providers\\RouteServiceProvider->boot()
#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#31 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#33 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call()
#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(1010): Illuminate\\Container\\Container->call()
#35 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(991): Illuminate\\Foundation\\Application->bootProvider()
#36 [internal function]: Illuminate\\Foundation\\Application->Illuminate\\Foundation\\{closure}()
#37 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(992): array_walk()
#38 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/BootProviders.php(17): Illuminate\\Foundation\\Application->boot()
#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(261): Illuminate\\Foundation\\Bootstrap\\BootProviders->bootstrap()
#40 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(447): Illuminate\\Foundation\\Application->bootstrapWith()
#41 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(199): Illuminate\\Foundation\\Console\\Kernel->bootstrap()
#42 /var/www/html/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#43 {main}
"} 
In Connection.php line 801:
                                                                               
  Database file at path [/var/www/html/database/database.sqlite] does not exi  
  st. Ensure this is an absolute path to the database. (Connection: sqlite, S  
  QL: PRAGMA foreign_keys = ON;)                                               
                                                                               
In SQLiteConnector.php line 34:
                                                                               
  Database file at path [/var/www/html/database/database.sqlite] does not exi  
  st. Ensure this is an absolute path to the database.                         
                                                                               
s6-rc: warning: unable to start service laravel-automations: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
prog: fatal: stopping the container.
```

**Comment by ChaseFreeman17** [comment]
Same issue here, I moved from latest tag back to `image: 'ghcr.io/alexjustesen/speedtest-tracker:v0.11.17'`

**Comment by Jiriteach** [comment]
I've ended up moving back as well to .17 and its working.

**Comment by alexdelprete** [comment]
> I've ended up moving back as well to .17 and its working.

are you using .env file? docker-compose? can you provide more info please? thanks.

in my case, .19 wasn't reading .env correctly, I moved all env variables to docker-compose file, and now everything's working fine. But I use MariaDB, not sqlite.

**Comment by Jiriteach** [comment]
```
version: '3.3'
services:
    speedtest-tracker:
        container_name: updated-speedtest-tracker

        network_mode: "host"

        environment:
            - PUID=1000
            - PGID=1000
            - TZ=Pacific/Auckland
            - 
        volumes:
            - '/home/pi/Docker/updated-speedtest-tracker/config:/config'
        image: 'ghcr.io/alexjustesen/speedtest-tracker:v0.11.17'

        restart: unless-stopped
```

**Comment by felho001** [comment]
same issue here, watchtower automatic update and I back to v0.11.17 and this is work

**Comment by SaltireSoul** [comment]
Looks like v0.11.18 removed sqlite support 
> [Chore] Remove support for sqlite by @alexjustesen in https://github.com/alexjustesen/speedtest-tracker/pull/669


Doesn't seem this statement was completely true.
> Also to note, this impacts new installs not existing.

_Originally posted by @alexjustesen in https://github.com/alexjustesen/speedtest-tracker/issues/669#issuecomment-1698451635_

**Comment by mrxxxxxx** [comment]
Same Issue. Use Sqlite

**Comment by felho001** [comment]
Hi how can I set the watchtower not updated speedtest tracking? It is possible? Because my set is 04:00am run all docker compose file updated.

**Comment by Jiriteach** [comment]
> Hi how can I set the watchtower not updated speedtest tracking? It is possible? Because my set is 04:00am run all docker compose file updated.

Set a label on the container -

com.centurylinklabs.watchtower.enable 

With a value of false

**Comment by felho001** [comment]
> > Hi how can I set the watchtower not updated speedtest tracking? It is possible? Because my set is 04:00am run all docker compose file updated.
> 
> Set a label on the container -
> 
> com.centurylinklabs.watchtower.enable 
> 
> With a value of false
I use one compose file and there many docker container where can I set? Speedtest tracker row?

**Comment by Christophe999s** [comment]
Same issue, went back to 0.11.17, which corrected the issue

**Comment by sahara101** [comment]
Same here. I will wait for Alex before going back :)

**Comment by Ghost108** [comment]
same issue. I solve it to swtich from sqlite file to mysql database.

**Comment by Drallas** [comment]
It's also failing the MariaDB version, also went back to 0.11.17.

**Comment by alexjustesen** [comment]
Oops... `v0.11.20` restores the symlink and is building now: https://github.com/alexjustesen/speedtest-tracker/releases/tag/v0.11.20

**Comment by MarkusMory** [comment]
I get the same error messages but the speedtest-tracker does work normally

**Comment by alexjustesen** [comment]
You'll need to pass an `env` variable stating the following and remove any other `DB_` environment variables you might be passing...

```
DB_CONNECTION=sqlite
```

**Comment by alexjustesen** [comment]
Pinning this issue and re-opening it to reduce duplicates until I can get the docs updated later today.

**Comment by CorneliousJD** [comment]
> You'll need to pass an `env` variable stating the following and remove any other `DB_` environment variables you might be passing...
> 
> ```
> DB_CONNECTION=sqlite
> ```
> 
> I'll update the docs later today with how to use sqlite as a DB just know going forward its intended for backwards compatibility only and the recommended DBs will be MySQL/MariaDB and Postresql.
> 
> Once #676 is done I'll be working on #591 so you can migrate from SQLite to any of the other options.

Adding the ENV variable you noted still results in an error for me 

In SQLiteConnector.php line 34:
                                                                               
  Database file at path [speedtest_tracker] does not exist. Ensure this is an  
   absolute path to the database.

**Comment by spupuz** [comment]
i'll have the same problem

**Comment by alexjustesen** [comment]
@CorneliousJD I'll take a closer look in a couple hours after meetings. Likely a result of some environment defaults.

**Comment by alexjustesen** [comment]
@CorneliousJD @spupuz can one of you try passing `DB_DATABASE=null` as an environment variable?

**Comment by Drallas** [comment]
Speedstest start normal, but getting a 500 error in the browser now. (using MariaDB)
When I change the tag back to '0.11.17' it's all good again.

**Comment by spupuz** [comment]
In Connection.php line 801:

                                                                               

  SQLSTATE[HY000] [14] unable to open database file (Connection: sqlite, SQL:  

   PRAGMA foreign_keys = ON;)                                                  

                                                                               

In Connector.php line 65:

                                                     

  SQLSTATE[HY000] [14] unable to open database file  

                                                     

s6-rc: warning: unable to start service laravel-automations: command exited 1

/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.

prog: fatal: stopping the container.

**Comment by AleksCee** [comment]
@Ghost108 
> same issue. I solved it to swtich from sqlite file to mysql database.
Did you only switch or migrate the data? And if you migrate, how? Only an sql-dump and import...?

**Comment by Ghost108** [comment]
I switched only. 
but yes you can make an sql dump and import this

**Comment by felho001** [comment]
New version not fix it I retired v0.11.17 version

**Comment by AleksCee** [comment]
> I switched only. but yes you can make a sql dump and import this

I have try it put the dump is not compatible for import in mariadb. :-(

**Comment by felho001** [comment]
Anyone know how can change my compose file work new version. 
container_name: speedtest-track
image: ghcr.io/alexjustesen/speedtest-tracker:v0.11.17
ports:
  - 8766:80
volumes:
  - /volume3/docker/appdata/speedtest-track:/config
environment:
  - TZ=Europe/Budapest
  - PGID=xx
  - PUID=xx
  - OOKLA_EULA_GDPR=true
  - TELEGRAM_BOT_TOKEN=1x
  - TELEGRAM_CHAT_ID=-x
network_mode: bridge
logging:
  driver: "json-file"
  options:
      max-file: "10"
      max-size: "200k"
restart: unless-stopped

**Comment by spupuz** [comment]
> Anyone know how can change my compose file work new version.
> 
> container_name: speedtest-track image: ghcr.io/alexjustesen/speedtest-tracker:latest ports:
> 
>     * 8766:80
>       volumes:
> 
>     * /volume3/docker/appdata/speedtest-track:/config
>       environment:
> 
>     * TZ=Europe/Budapest
> 
>     * PGID=xx
> 
>     * PUID=xx
> 
>     * OOKLA_EULA_GDPR=true
> 
>     * TELEGRAM_BOT_TOKEN=1x
> 
>     * TELEGRAM_CHAT_ID=-x
>       network_mode: bridge
>       logging:
>       driver: "json-file"
>       options:
>       max-file: "10"
>       max-size: "200k"
>       restart: unless-stopped

No latest version seems not working

**Comment by spupuz** [comment]
> Speedstest start normal, but getting a 500 error in the browser now. (using MariaDB) When I change the tag back to '0.11.17' it's all good again.
> 
> **Normal startup** <img alt="Screenshot 2023-08-31 at 15 59 18" width="777" src="https://user-images.githubusercontent.com/24792888/264671517-dbb0681c-ed5e-465c-bcce-1ca6d717137a.png">
> 
> **Can't get into the WebGUI Error 500 in Browser** <img alt="Screenshot 2023-08-31 at 15 59 24" width="570" src="https://user-images.githubusercontent.com/24792888/264671564-2ded5c0c-302a-439f-8d87-d1c4e5a01d8c.png">
> 
> **But Data via API.** <img alt="Screenshot 2023-08-31 at 15 59 34" width="460" src="https://user-images.githubusercontent.com/24792888/264671671-7a9e68dd-fa04-4242-b7f5-5c40c7e8136a.png">

qhich tool are you using for container logs?

**Comment by Drallas** [comment]
> > Speedstest start normal, but getting a 500 error in the browser now. (using MariaDB) When I change the tag back to '0.11.17' it's all good again.
> > **Normal startup** <img alt="Screenshot 2023-08-31 at 15 59 18" width="777" src="https://user-images.githubusercontent.com/24792888/264671517-dbb0681c-ed5e-465c-bcce-1ca6d717137a.png">
> > **Can't get into the WebGUI Error 500 in Browser** <img alt="Screenshot 2023-08-31 at 15 59 24" width="570" src="https://user-images.githubusercontent.com/24792888/264671564-2ded5c0c-302a-439f-8d87-d1c4e5a01d8c.png">
> > **But Data via API.** <img alt="Screenshot 2023-08-31 at 15 59 34" width="460" src="https://user-images.githubusercontent.com/24792888/264671671-7a9e68dd-fa04-4242-b7f5-5c40c7e8136a.png">
> 
> qhich tool are you using for container logs?

That’s Portainer logging.

**Comment by mvanlijden** [comment]
Just did a fresh install on Unraid based on template and included the DB_CONNECTION variable for sqlite but still not working.

```
In Connection.php line 801:
                                                                               
  Database file at path [speedtest_tracker] does not exist. Ensure this is an  
   absolute path to the database. (Connection: sqlite, SQL: PRAGMA foreign_ke  
  ys = ON;)                                                                    
                                                                               

In SQLiteConnector.php line 34:
                                                                               
  Database file at path [speedtest_tracker] does not exist. Ensure this is an  
   absolute path to the database. 
```

**Comment by chandz05** [comment]
> Just did a fresh install on Unraid based on template and included the DB_CONNECTION variable for sqlite but still not working.
> 
> ```
> In Connection.php line 801:
>                                                                                
>   Database file at path [speedtest_tracker] does not exist. Ensure this is an  
>    absolute path to the database. (Connection: sqlite, SQL: PRAGMA foreign_ke  
>   ys = ON;)                                                                    
>                                                                                
> 
> In SQLiteConnector.php line 34:
>                                                                                
>   Database file at path [speedtest_tracker] does not exist. Ensure this is an  
>    absolute path to the database. 
> ```

My db connection issues and 500 refused errors have been resolved on docker in UnRaid. For Community Applications speedtest-tracker image, I had to explicitly add DB details as env variables as well as the APP_KEY variable. Previously this was not required

**Comment by chandz05** [comment]
> can you share the env variable and your anonymized docker compose pls

Absolutely I can. Please note this is the UnRaid managed Docker Run script:

`docker run
  -d
  --name='speedtest-tracker'
  --net='bridge'
  -e TZ="America/New_York"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="Tower"
  -e HOST_CONTAINERNAME="speedtest-tracker"
  -e 'PUID'='1000'
  -e 'PGID'='1000'
  -e 'APP_KEY'='base64:somstring'
  -e 'DB_CONNECTION'='mysql'
  -e 'DB_HOST'='db'
  -e 'DB_PORT'='3306'
  -e 'DB_DATABASE'='speedtest'
  -e 'DB_USERNAME'='speedtest'
  -e 'DB_PASSWORD'='password'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='https://[IP]:[PORT:443]/'
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/devzwf/unraid-docker-templates/main/images/speedtest-tracker-icon.png'
  -p '9443:443/tcp'
  -v '/mnt/user/appdata/speedtest-tracker-1':'/config':'rw' 'ghcr.io/alexjustesen/speedtest-tracker:latest'`

To get the APP_KEY value, run the following command for your speedtest-tracker container:
`docker exec -it speedtest-tracker php artisan key:generate --show`

**Comment by spupuz** [comment]
uh ok but you are using myql i tought you where using sqllite

**Comment by chandz05** [comment]
> uh ok but you are using myql i tought you where using sqllite

Sure, but it seems like a big issue is that the application cannot read the variables in the .env file anymore. Specify your sqllite DB as well as APP_KEY in your docker compose/run files and see if that works. If not, then it is a different issue

**Comment by holocronology** [comment]
I tried fresh clean installs and get similar errors with MariaDB

**Comment by Jiriteach** [comment]
@alexjustesen - Tried with DB_DATABASE=null and DB_DATABASE=sqlite and no go.

Also tried with no existing config.

Still errors with cannot find the db.

Have dropped back to v17 for now.

**Comment by spupuz** [comment]
> @alexjustesen - Tried with DB_DATABASE=null and DB_DATABASE=sqlite and no go.
> 
> Also tried with no existing config.
> 
> Still errors with cannot find the db.
> 
> Have dropped back to v17 for now.

Same situation here

**Comment by spupuz** [comment]
We need an explaination

**Comment by Jiriteach** [comment]
> @holocronology @Jiriteach Have you guys tried adding the APP_KEY variable?

I've seen the other thread but my errors are not about the APP_KEY - its cannot find the db. I'll try and see if does make a diff. I dont use a .env file.

**Comment by holocronology** [comment]
> @holocronology @Jiriteach Have you guys tried adding the APP_KEY variable?

Yes, I did. Similar issues. I also noticed that the rest of the DB info (host, port, etc) was not being properly passed from compose file to mariaDB.

**Comment by Jiriteach** [comment]
My issue of db not found is not related to the APP_KEY issue. Flushed everything and a new install failed - seems like a path issue in being able to find the sqlite db. Have reverted to .17 and using this OK for now.

**Comment by EVOTk** [comment]
> Just did a fresh install on Unraid based on template and included the DB_CONNECTION variable for sqlite but still not working.
> 
> ```
> In Connection.php line 801:
>                                                                                
>   Database file at path [speedtest_tracker] does not exist. Ensure this is an  
>    absolute path to the database. (Connection: sqlite, SQL: PRAGMA foreign_ke  
>   ys = ON;)                                                                    
>                                                                                
> 
> In SQLiteConnector.php line 34:
>                                                                                
>   Database file at path [speedtest_tracker] does not exist. Ensure this is an  
>    absolute path to the database. 
> ```

Same issue for me :(

**Comment by Jiriteach** [comment]
Decided to try 0.11.21 and strangly - its fixed!. Tried this on multiple sites with the 0.11.21 version. Fails on first startup then restarts and fixed. Thanks

**Comment by EVOTk** [comment]
> Decided to try 0.11.21 and strangly - its fixed!. Tried this on multiple sites with the 0.11.21 version. Fails on first startup then restarts and fixed so I belive its fixed now. Thanks

Same for me, fails on first startup then restarts , its works

**Comment by StirCwazy** [comment]
My apologies if this is somewhere mixed into this huge gnarly thread, but is 0.11.21 working on unRAID?  Default install, no existing appdata, and I'm getting the following.

In Connection.php line 801:
                                                                               
  SQLSTATE[HY000] [2002] Connection refused (Connection: mysql, SQL: select *  
   from information_schema.tables where table_schema = speedtest_tracker and   
  table_name = migrations and table_type = 'BASE TABLE')                       
                                                                               

In Connector.php line 65:
                                             
  SQLSTATE[HY000] [2002] Connection refused

This is a clean install on a new server, had this on a previous server a while back and wanted to start using it again.  I came across this thread and have seen numerous comments about unRAID and it not working.

**Comment by chandz05** [comment]
@StirCwazy this is what works for me on Uraid. Note that I setup a separate db for speed test:

> -d
> --name='speedtest-tracker'
> --net='bridge'
> -e TZ="America/New_York"
> -e HOST_OS="Unraid"
> -e HOST_HOSTNAME="Tower"
> -e HOST_CONTAINERNAME="speedtest-tracker"
> -e 'PUID'='1000'
> -e 'PGID'='1000'
> -e 'APP_KEY'='base64:somstring'
> -e 'DB_CONNECTION'='mysql'
> -e 'DB_HOST'='db'
> -e 'DB_PORT'='3306'
> -e 'DB_DATABASE'='speedtest'
> -e 'DB_USERNAME'='speedtest'
> -e 'DB_PASSWORD'='password'
> -l net.unraid.docker.managed=dockerman
> -l net.unraid.docker.webui='https://[IP]:[PORT:443]/'
> -l net.unraid.docker.icon='https://raw.githubusercontent.com/devzwf/unraid-docker-templates/main/images/speedtest-tracker-icon.png'
> -p '9443:443/tcp'
> -v '/mnt/user/appdata/speedtest-tracker-1':'/config':'rw' 'ghcr.io/alexjustesen/speedtest-tracker:latest'

To get the APP_KEY value, run the following command for your speedtest-tracker container:
`docker exec -it speedtest-tracker php artisan key:generate --show`

**Comment by holocronology** [comment]
UNRAID: I had it on v0.11.17. I changed the template back to "latest". Forced an update and it failed to start. Same errors. Stopped and restarted and it works.

**Comment by StirCwazy** [comment]
> @StirCwazy this is what works for me on Unraid. Note that I setup a separate db for speed test:
> 
> > -d
> > --name='speedtest-tracker'
> > --net='bridge'
> > -e TZ="America/New_York"
> > -e HOST_OS="Unraid"
> > -e HOST_HOSTNAME="Tower"
> > -e HOST_CONTAINERNAME="speedtest-tracker"
> > -e 'PUID'='1000'
> > -e 'PGID'='1000'
> > -e 'APP_KEY'='base64:somstring'
> > -e 'DB_CONNECTION'='mysql'
> > -e 'DB_HOST'='db'
> > -e 'DB_PORT'='3306'
> > -e 'DB_DATABASE'='speedtest'
> > -e 'DB_USERNAME'='speedtest'
> > -e 'DB_PASSWORD'='password'
> > -l net.unraid.docker.managed=dockerman
> > -l net.unraid.docker.webui='https://[IP]:[PORT:443]/'
> > -l net.unraid.docker.icon='https://raw.githubusercontent.com/devzwf/unraid-docker-templates/main/images/speedtest-tracker-icon.png'
> > -p '9443:443/tcp'
> > -v '/mnt/user/appdata/speedtest-tracker-1':'/config':'rw' 'ghcr.io/alexjustesen/speedtest-tracker:latest'
> 
> To get the APP_KEY value, run the following command for your speedtest-tracker container: `docker exec -it speedtest-tracker php artisan key:generate --show`

Am I understanding then that a sqlite DB is no longer an option?

**Comment by alexjustesen** [comment]
> Decided to try 0.11.21 and strangly - its fixed!. Tried this on multiple sites with the 0.11.21 version. Fails on first startup then restarts and fixed so I belive its fixed now. Thanks

I added back the symlink in `0.11.20` so that fixed the connection to the database because it needed to get mounted inside of the app which is why it suddenly started work.

**Comment by SaltireSoul** [comment]
> > Decided to try 0.11.21 and strangly - its fixed!. Tried this on multiple sites with the 0.11.21 version. Fails on first startup then restarts and fixed so I belive its fixed now. Thanks
> 
> I added back the symlink in `0.11.20` so that fixed the connection to the database because it needed to get mounted inside of the app which is why it suddenly started work.

It is still kind of bugged, as it requires two runs to start successfully.

**Comment by alexjustesen** [comment]
> I guess the Unraid APP template is going to need to be revised to include mariadb/mysql or postsql databases.

I don't own the template but I'll post in the forums for an update.

**Comment by alexjustesen** [comment]
> It is still kind of bugged, as it requires two runs to start successfully.

Caching I think has something to do with it

**Comment by SaltireSoul** [comment]
I forgot to mention, ZappyZap's Unraid template doesn't use enviroment variables except for `/config`

**Comment by alexjustesen** [comment]
Maybe not by default but you can add them 
![image](https://github.com/alexjustesen/speedtest-tracker/assets/1144087/39cb4350-75ab-483f-a28d-ffa77a3d7067)

**Comment by SaltireSoul** [comment]
@StirCwazy I'm actually using Unraid for this myself, however I'm using the `.env` file.
```
DB_CONNECTION=sqlite
DB_DATABASE=/config/database.sqlite
```

@alexjustesen sorry I just meant it creates a `.env` file by default and manually creating enviroment variables might be having a conflict

**Comment by alexjustesen** [comment]
> @alexjustesen sorry I just meant it creates a `.env` file by default and manually creating enviroment variables might be having a conflict

No worries, I'm marching towards getting rid of the the `.env` file so environment variables passed through are absolutely preferred at this point.


> @alexjustesen If using unraid, are there any other variables I'd need to use if opting for sqlite?

What @SaltireSoul is correct, you just need those two vars.

**Comment by holocronology** [comment]
Adding these to Unraid:
![image](https://github.com/alexjustesen/speedtest-tracker/assets/65265114/3b173617-a37e-4cf7-9cab-9a9b8670868a)

Resulted in:
![image](https://github.com/alexjustesen/speedtest-tracker/assets/65265114/18132e21-8c2f-43ff-9254-7c64898e0d9a)

**Comment by StirCwazy** [comment]
> Adding these to Unraid: ![image](https://user-images.githubusercontent.com/65265114/265560733-3b173617-a37e-4cf7-9cab-9a9b8670868a.png)
> 
> Resulted in: ![image](https://user-images.githubusercontent.com/65265114/265560766-18132e21-8c2f-43ff-9254-7c64898e0d9a.png)
> 
> Removed them. Started and it failed. Restarted and it worked.

Added both variables, got the same 500 | Server Error.  Removed as you said you tried with success, failed.

In Connection.php line 801:
                                                                               
  SQLSTATE[HY000] [2002] Connection refused (Connection: mysql, SQL: select *  
   from information_schema.tables where table_schema = speedtest_tracker and   
  table_name = migrations and table_type = 'BASE TABLE')                       
                                                                               

In Connector.php line 65:
                                             
  SQLSTATE[HY000] [2002] Connection refused

**Comment by alexjustesen** [comment]
@StirCwazy config looks right, not sure if it was the unraid forums or here but it required a second restart for the container to pickup the env variables.

**Comment by holocronology** [comment]
> it required a second restart for the container to pickup the env variables

I did try the second, and third, restart after adding the variables without success. It was only after removing them that it worked.

**Comment by devzwf** [comment]
> speedtest_tracker



> @StirCwazy config looks right, not sure if it was the unraid forums or here but it required a second restart for the container to pickup the env variables.

That's really weird , but yes it is the case.

**Comment by StirCwazy** [comment]
> > speedtest_tracker
> 
> > @StirCwazy config looks right, not sure if it was the unraid forums or here but it required a second restart for the container to pickup the env variables.
> 
> That's really weird , but yes it is the case.

After adding the variables for DB_CONNECTION and DB_DATABASE it worked after a restart.  I've been busy and haven't had a chance to reply, so my apologies to all.

**Comment by alexjustesen** [comment]
#746 merged to dynamically generate an `APP_KEY` should that be missing on start up.

**Comment by toxic0berliner** [comment]
Seems it's not creating the db if it doesn't exist. How to start using this image when no preexisting sqlite db exists? Should be documented I believe

**Comment by alexjustesen** [comment]
[Using SQLite](https://docs.speedtest-tracker.dev/faqs#using-sqlite) added to the FAQ

**Comment by felho001** [comment]
> [Using SQLite](https://docs.speedtest-tracker.dev/faqs#using-sqlite) added to the FAQ

I try this method but not work for me. I back to v0.11.17

**Comment by holocronology** [comment]
I moved to MariaDB.

**Comment by alexjustesen** [comment]
> > [Using SQLite](https://docs.speedtest-tracker.dev/faqs#using-sqlite) added to the FAQ
> 
> I try this method but not work for me. I back to v0.11.17

If you're still having problems please open a new issue and we'll get it sorted. I'll be unsubscribing from this one
